Permafail E2E: redesign the Help Centre spec.

worldomonation opened the pull request:
Fixes https://github.com/Automattic/wp-calypso/issues/78358.

## Proposed Changes

This PR rethink the Support (now renamed Help Centre) specs.

Key changes:
- remove the spec that tests various inputs. [This is better served as a component test](https://github.com/Automattic/wp-calypso/issues/77724).
- rename the specs to reference Help Center (US spelling).
- refactor the remaining spec to simulate a user looking for information on a topic.

## Testing Instructions

Ensure the following build configurations are passing:
  - [x] Calypso E2E (desktop)
  - [x] Calypso E2E (mobile)

100 iterations survived on desktop viewport:
![image](https://github.com/Automattic/wp-calypso/assets/6549265/9e0110c7-b415-4892-9053-4b850d9ec2c8)

100 iterations survived on mobile viewport:
![image](https://github.com/Automattic/wp-calypso/assets/6549265/03134225-21ac-428e-8dd4-5a1b4549eaa5)


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

worldomonation committed:
test/e2e/specs/support/support__popover.ts
- remove spec in favor of a (yet-to-be-written) component test.

test/e2e/specs/support/support__showme-where.ts
- renamed.

test/e2e/specs/help-center/help-center__general-interaction.ts
- refactored to run through the support help docs, then the calypso link.

packages/calypso-e2e/src/lib/components/support-component.ts
- use accessible selectors.
- remove methods no longer used.

karenroldan commented on packages/calypso-e2e/src/lib/components/support-component.ts:
```diff
@@ -1,11 +1,13 @@
+
 /**
- * Represents the Inline Help popover.
+ * Represnts the Help Center popover.
```
**typo:** Represnts > Represents

worldomonation replied:
Fixed in 54fadb4fceb68c779c64ccc6f287e0daa2e7acb3.

karenroldan commented on test/e2e/specs/help-center/help-center__general-interaction.ts:
```diff
@@ -0,0 +1,65 @@
+		await testAccount.authenticate( page );
+
+		supportComponent = new SupportComponent( page );
+		await page.waitForLoadState( 'networkidle' );
```
**thought:** I couldn't remember exactly if we're going to avoid using `networkidle` or not (or if there is an ongoing discussion about this).

worldomonation replied:
You remember correct! It's [not recommended anymore](https://playwright.dev/docs/api/class-page#page-go-back-option-wait-until). 

I used it as a temporary placeholder until the [PR with changes to the TestAccount.authenticate method](https://github.com/Automattic/wp-calypso/pull/78616/files#r1243116632) was merged, and then was going to rebase. Sorry for not making this fact clear.

karenroldan commented on packages/calypso-e2e/src/lib/components/support-component.ts:
```diff
@@ -65,28 +67,53 @@ export class SupportComponent {
+	 * @param {ResultsCategory} category Category of results to click on.
+	 * @param {number} index Locate results based on index. 0-indexed.
+	 */
+	async clickResultByIndex( category: ResultsCategory, index: number ) {
```
**thought (non-blocking):** I like the flexibility that the `index` parameter gives, but I'm pondering over its necessity, especially if we're only interacting with the first result.

worldomonation replied:
> I'm pondering over its necessity, especially if we're only interacting with the first result.

Totally fair point!

I prefer to make methods applicable in a general context over singular-purpose, which is why I ended up implementing the method I did instead of something like, say, `clickFirstResult`.

This method is also a leftover from a slightly more ambitious spec I wrote then scrapped for being unworkable, where I wanted to verify that I can match on both index and title. As you can see, it didn't end up working out.

I'll change the spec to interact with something other than the first item so the method doesn't seem so out of place.

worldomonation replied:
Revised in 54fadb4fceb68c779c64ccc6f287e0daa2e7acb3

karenroldan reviewed (approved):
LGTM! 🚀

worldomonation committed:
Fix issues pointed out in review.

packages/calypso-e2e/src/lib/components/support-component.ts
- fix typo.

test/e2e/specs/help-center/help-center__general-interaction.ts
- select the second help doc instead of first.

worldomonation committed:
test/e2e/specs/help-center/help-center__general-interaction.ts
- use changes introduced in https://github.com/Automattic/wp-calypso/pull/78616.

worldomonation committed:
test/e2e/specs/help-center/help-center__general-interaction.ts
- add step to clear results.

worldomonation committed:
packages/calypso-e2e/src/lib/components/support-component.ts
- add workaround for mobile by checking whether the Close Search button is present.
